tests/ui/impl-trait/non-defining-uses/double-wrap-with-defining-use.rs

```diff
@@ -1,12 +1,17 @@
 // Regression test for ICE from issue #140545
 // The error message is confusing and wrong, but that's a different problem (#139350)
+
 //@ edition:2018
+//@ revisions: current next
+//@[next] compile-flags: -Znext-solver
+//@ ignore-compare-mode-next-solver (explicit revisions)
+//@[next] check-pass
 
 trait Foo {}
 fn a(x: impl Foo) -> impl Foo {
```

Comment:
does this need to be an APIT? seems more confusing than a normal type parameter

also there is no nested bodies here. is this just a drive-by increasing test coverage or did this prev fail with new solveR?